lib/workers/repository/updates/branch-name.ts

```diff
@@ -56,7 +56,7 @@ export function generateBranchName(update: RenovateConfig): void {
     update.groupSlug = slugify(update.groupSlug ?? update.groupName, {
       lower: true,
     });
-    if (update.updateType === 'major' && update.separateMajorMinor) {
+    if (update.updateType === 'major' && update.separateMajorMinor !== false) {
```

Comment:
What's the impact of this change? It will now evaluate to truthy if it's `undefined` or `null`, but is that useful?

Reply:
the docs say to set it to false to disable, so it assume it's enabled by default. but effectively it's disabled by default, because sinve this change it only separates when the option is explicit enabled.

The option is currently undefined by default. I can also send a fix pr to update the docs, that we need to explicit enable, so no breaking change. just a docs fix.
Then we can do this breaking in a new major version.

Reply:
**Docs:**
![image](https://user-images.githubusercontent.com/1798109/204255274-0e2aadf2-38c6-4f55-aade-14e67db63bdd.png)

Reply:
For me this is a bug in behavior, as the docs clearly say it's enabled by default, but isn't until this PR.

Reply:
Do we know which types of branches this will change as a result?

Reply:
yes, major updates are getting the `major-` prefix as expected. so probably new major update PR's are occuring. also maybe some will get closed because of the branch name fix.

Reply:
But for example every major PR will get autoclosed?

Reply:
probably yes, if `separateMajorMinor` is not explicitly set to true